### PR TITLE
openhie demo workflow

### PR DESCRIPTION
**Pull request by mtuchi**
### Notes for the reviewer 

This PR address the request in #3 , I have updated the required jobs for the demo and run the jobs on [app.openfn.org](https://app.openfn.org/projects/fdfdf286-aa8e-4c9e-a1d2-89c1e6928a2a/runs/d8aa35d9-814a-4113-b7b2-5aff7fe47651)

Something to note when testing the job locally using the `@openfn/cli` `workflow.json`. There is a bug in runtime when passing the final state of the upstream job to two downstream jobs. Eg The final state of `validate`, Is supposed to be passed to `convert` and `dhis2` but at the moment `dhis2` job doesn't get its initial state from `validate` and hence it will fail when running it locally. 

@josephjclark is working on a fix, that it will make sure that if there are two downstream jobs that needs the final state of the upstream job, They will both use the final state of that upstream job as their initial state

### CODE REVIEW

File: lightning/send-to-hapi.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+fn(state => {
+  return post('Patient', state.patient, state => {
+    const { encounter } = state;
+
+    encounter.body.subject.reference = `Patient/${state.data.id}`;
+    return { ...state, encounter };
+  })(state);
+});
+
+fn(state => {
```
Comment (taylordowns2000) [inline lightning/send-to-hapi.js:16]: this looks like an antipattern. go with `post('Encounter', state => state.encounter)` instead... no need for this `fn()` block

File: lightning/send-to-google.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fn(state => {
+  const { patient, encounter } = state;
+  console.log('Patient', patient);
+  console.log('Encounter', encounter);
+  return state;
+});
+
+// post("blah.com")
```
Comment (taylordowns2000) [inline lightning/send-to-google.js:8]: please add a TODO in here to connect to these APIs: https://cloud.google.com/healthcare-api/docs/concepts/fhir#fhir_stores

File: lightning/send-to-hapi.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  const { patient, encounter } = state.data;
+
+  return { ...state, patient, encounter };
+});
+
+fn(state => {
```
Comment (taylordowns2000) [inline lightning/send-to-hapi.js:7]: the dreaded custom execute anti-pattern here! please remove

File: lightning/send-to-hapi.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
```
Comment (taylordowns2000) [inline lightning/send-to-hapi.js:1]: delete this if no longer needed

**Review by taylordowns2000 — CHANGES_REQUESTED** [review]
i've requested 4 changes via inline comments. beyond those, please implement this on app.openfn.org and let me know when done! we'll merge it in 😄

**Comment by josephjclark** [comment]
> @josephjclark is working on a fix, that it will make sure that if there are two downstream jobs that needs the final state of the upstream job, They will both use the final state of that upstream job as their initial state

I am looking at a fix, it needs a bit of thought. But it's not blocking!

a) Lightning doesn't use the CLI's workflows yet (won't for a while)
b) It's quite reasonable to return state at the end of each job.

To be fair, as this is a demo, we should also be encouraging better practice of state. We WANT users to write all shared state to `state.data`. You shouldn't need to write code to promote it to the top of the state object - just let it sit on data.

So yes, I'm looking into the issues, but a cleaner version of this should be compatible with Lightning, the  CLI, and the pattersn we're trying to encourage :)

**Pull request merged**